configs/raspberrypi_config

```diff
@@ -78,25 +83,40 @@ function platform_modify() {
         run_and_log_cmd "sudo cp work/boot/${RASPBERRYPI_KERNEL_IMAGE} work/rootfs/boot/${MENDER_KERNEL_IMAGETYPE}"
     fi
 
+    # Make sure the on-board bootloader boots our bootloader (which we do via the kernel option).
+    sed -i -e '/^kernel=/d' work/boot/config.txt
+    echo "kernel=${RASPBERRYPI_BOOTLOADER_IMAGE}" >> work/boot/config.txt
+
     # Replace kernel with U-boot and add boot script
     run_and_log_cmd "sudo mkdir -p work/rootfs/uboot"
     run_and_log_cmd "sudo cp work/rpi/binaries/u-boot.bin work/boot/${RASPBERRYPI_BOOTLOADER_IMAGE}"
     run_and_log_cmd "sudo cp work/rpi/binaries/boot.scr work/boot"
     run_and_log_cmd "sudo cp work/rpi/binaries/fw_env.config work/rootfs/etc/"
     run_and_log_cmd "sudo cp work/rpi/binaries/uboot-git-log.txt work/boot"
 
+    # Disable 64-bit mode and force 32-bit mode. Even though we check for 32-bit binaries at the
+    # start of this function, some images use a 64-bit kernel even with 32-bit userspace. Due to our
+    # custom U-Boot bootloader we need to stick to 32-bit for now.
+    if ! grep -q '^arm_64bit=0' work/boot/config.txt; then
+        log_info "Forcing 32-bit kernel"
+        sed -i -e '/^arm_64bit=/d' work/boot/config.txt
+        echo "arm_64bit=0" >> work/boot/config.txt
+    fi
+
     # Raspberry Pi applications expect to find this on the device and in some
     # cases parse the options to determine the functionality. Create symlinks
-    # to all files
+    # to all files that don't already exist.
     for f in $(ls -1 work/boot); do
-        run_and_log_cmd "sudo ln -fs /uboot/${f} work/rootfs/boot/${f}"
+        if [ ! -e "work/rootfs/boot/${f}" ]; then
+            run_and_log_cmd "sudo ln -fs /uboot/${f} work/rootfs/boot/${f}"
```

Comment:
I do not understand the ln command here, `/uboot/${f}` is the target, and link name is `work/rootfs/boot/${f}`, just please double check if thats what we want here (I lack the knowledge). also if you decide to go along with the find, remember that it will contain the full path, so you have to use basename appropriately.

Reply:
`/uboot` is the boot partition, and `/boot` is on the root partition (I know, confusing, but that's how it is). We want to link to some files (most importantly, dtb files) on the boot partition, so this should indeed be correct: We link files in `/boot` to `/uboot`. This isn't a very clever integration, because it increases the reliance on the boot partition, which isn't updated in rootfs-image updates, but it's not something I want to change now.